### ISSUE TITLE
IDE freeze for a project with multiples modules and overrides

### ISSUE THREAD

**Issue by DiegoCoronel** [ceylon/ceylon-ide-intellij#430]
I Tried to configure this [project](https://github.com/DiegoCoronel/ceylon-spring-boot) but it takes too long to respond.. also, it was the only project i wasnt able to configure in Intellij

**Comment by davidfestal** [comment]
Since this project has *many* binary dependencies to download on the internet, I assume the freeze was due to https://github.com/ceylon/ceylon-ide-intellij/issues/441

**Comment by DiegoCoronel** [comment]
Ill try again tomorrow

**Comment by bjansen** [comment]
I confirm this is related to #441. I have a fix that will soon be pushed, and when I open your project, the UI doesn't freeze anymore, but:

* the first time I open the project, it takes a long time to download all the maven dependencies because I don't have them in my local repo (but that's expected)
* the project contains 5 sub-projects, so for each one of them IDEA will instantiate a new Ceylon model (the status bar will show `Ceylon model update`)
* for each Ceylon model, IDEA will try to attach sources to every dependency, but it looks like a lot of dependencies don't have any corresponding source in Maven Central, so it takes quite a long time "for nothing"

To sum up, loading this projects can take a significant amount of time, but the UI should never freeze anymore, and you can see what's happening in the bottom status bar by clicking on the "Ceylon model update" task.

To have faster loading times, you could merge the 5 sub-projects in a single project with multiple modules. AFAIU, this is a single application split in multiple modules, so the perfect use-case for Ceylon modules grouped under the same `source` directory ;)

**Comment by DiegoCoronel** [comment]
Thanks, much better now :)
